src/platform/api/pythonApi.ts

```diff
@@ -411,14 +410,14 @@ export class InterpreterService implements IInterpreterService {
         if (!this.extensionChecker.isPythonExtensionInstalled) {
             return;
         }
-        if (!this.api) {
```

Comment:
It's only caching the result here. Not the actual promise. So this could be called over and over and all of the calls would update this.api.  In the test one of them would be disabled.